Don't stage encrypted AIPs twice, refs #1112

djjuhasz opened the pull request:
Connects to https://github.com/archivematica/Issues/issues/1112

Update `self.current_location` to the temporary staging directory after
copying and decrypting the package.  This is necessary so subsequent
calls to `fetch_local_path()` check if the current (staging) Space is
encrypted (via `is_encrypted()`), not whether the AIPsStore Space
is encrypted.  If `fetch_local_path()` thinks the package is still
encrypted it will copy it to a new temp dir and decrypt it again.

Also:
- Copy encrypted AIPs directly from storage location to temporary
  staging directory and decrypt there
- Add a debug log message to `get_download_path()`

djjuhasz committed:
Don't stage encrypted AIPs twice, refs #1112

Update `self.current_location` to the temporary staging directory after
copying and decrypting the package.  This is necessary so subsequent
calls to `fetch_local_path()` check if the current (staging) Space is
encrypted (via `is_encrypted()`), not whether the AIPsStore Space
is encrypted.  If `fetch_local_path()` thinks the package is still
encrypted it will copy it to a new temp dir and decrypt it again.

Also:
- Copy encrypted AIPs directly from storage location to temporary
  staging directory and decrypt there
- Add a debug log message to `get_download_path()`

sevein reviewed (commented):
> Update self.current_location to the temporary staging directory after
copying and decrypting the package. This is necessary so subsequent
calls to fetch_local_path() check if the current (staging) Space is
encrypted (via is_encrypted()), not whether the AIPsStore Space
is encrypted. If fetch_local_path() thinks the package is still
encrypted it will copy it to a new temp dir and decrypt it again.

David, in regards to this specific comment, I'm trying to reproduce the scenario that you describe but I couldn't figure out the context where subsequent calls to fetch_local_path() are happening within the same web request. How could I reproduce?

djjuhasz commented:
> David, in regards to this specific comment, I'm trying to reproduce the scenario that you describe but I couldn't figure out the context where subsequent calls to fetch_local_path() are happening within the same web request. How could I reproduce?

@sevein I don't remember the exact details, as I haven't looked at the code for almost two months, but I followed the "Steps to reproduce" from Ashley's bug report https://github.com/archivematica/Issues/issues/1112 to reproduce the bug and test the fix.

sevein committed:
Make is_compressed look up cached local_path first

This commit ensures that is_compressed (property getter) looks up the local
path that may have been cached previously before. This is useful in the context
of `get_download_path` to avoid calling fetch_local_path twice which is
expensive.

sevein reviewed (approved):
LGTM.

sevein commented:
31729ca1d852d394b4f5168dba47bd9ca257670b and 056f71231c30766dcc253820da6bea6a0929a929 in stable/0.18.x
